In [29]:
import pandas as pd
import scipy as sp
import numpy as np
from scipy.stats import norm

# Cutlets

In [3]:
cutlets = pd.read_csv("C://Users//Lenovo//Downloads//Cutlets.csv")

In [8]:
cutlets.describe()

Unit A     Unit B
count  35.000000  35.000000
mean    7.019091   6.964297
std     0.288408   0.343401
min     6.437600   6.038000
25%     6.831500   6.753600
50%     6.943800   6.939900
75%     7.280550   7.195000
max     7.516900   7.545900

In [11]:
cutlets.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 35 entries, 0 to 34
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Unit A  35 non-null     float64
 1   Unit B  35 non-null     float64
dtypes: float64(2)
memory usage: 688.0 bytes


In [16]:
x1 = norm.rvs(loc=7.019,scale= 0.288, size = 35)
x1

array([7.46317006, 7.33080052, 6.46010742, 6.64608738, 6.88276968,
       6.976416  , 6.8230607 , 6.97571811, 7.304683  , 7.42393437,
       6.73774085, 6.60448414, 7.28690789, 7.11571255, 6.84849927,
       7.19231607, 6.83776931, 6.85393471, 6.87033683, 6.85568145,
       7.57690892, 7.17968039, 7.16690239, 6.92409644, 7.35710422,
       6.5337345 , 7.22925051, 7.14790635, 7.08620145, 6.63015503,
       7.14523294, 7.22361421, 7.0963974 , 7.20989041, 6.74310358])

In [17]:
x2 = norm.rvs(loc=6.96,scale=0.34,size=35)
x2

array([7.08196291, 7.25779832, 7.14032737, 7.07592522, 6.83271871,
       6.39895964, 6.75744321, 6.49448648, 6.31657   , 7.17671955,
       6.8575193 , 6.88463598, 6.78979585, 7.31230521, 7.13265836,
       6.97572023, 7.70307316, 7.18879321, 6.76860555, 6.76679472,
       6.68255022, 7.38877511, 7.81756734, 7.31324513, 7.09911386,
       6.65439565, 6.84133934, 6.43115021, 6.22930386, 7.20539955,
       6.472186  , 7.3076467 , 7.73918643, 7.14893827, 6.73535054])

In [18]:
stats.ttest_ind(x1,x2)

Ttest_indResult(statistic=0.6172774427340942, pvalue=0.5391132950425765)

p>0.05 ---> We accept null hypothesis , i.e. No changes is required

# LabTAT.mtw

In [4]:
lab = pd.read_csv("C://Users//Lenovo//Downloads//labTAT.csv")

In [5]:
lab.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [6]:
lab.describe()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
count    120.000000    120.000000    120.000000     120.00000
mean     178.361583    178.902917    199.913250     163.68275
std       13.173594     14.957114     16.539033      15.08508
min      138.300000    140.550000    159.690000     124.06000
25%      170.335000    168.025000    188.232500     154.05000
50%      178.530000    178.870000    199.805000     164.42500
75%      186.535000    189.112500    211.332500     172.88250
max      216.390000    217.860000    238.700000     205.18000

In [8]:
lab.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 120 entries, 0 to 119
Data columns (total 4 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Laboratory 1  120 non-null    float64
 1   Laboratory 2  120 non-null    float64
 2   Laboratory 3  120 non-null    float64
 3   Laboratory 4  120 non-null    float64
dtypes: float64(4)
memory usage: 3.9 KB


In [12]:
import scipy.stats as stats
tstat,pvalue = stats.f_oneway(lab.iloc[:,0], lab.iloc[:,1],lab.iloc[:,2],lab.iloc[:,3])


p<0.05, Reject the hypothesis
There is a significant difference between the groups' mean Lab values.

# Buyer Ratio.mtw

In [16]:
buyer = pd.read_csv("C://Users//Lenovo//Downloads//BuyerRatio.csv")

In [17]:
buyer.head()

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [18]:
buyer.describe()

East         West        North       South
count    2.000000     2.000000     2.000000    2.000000
mean   242.500000   832.500000   743.500000  410.000000
std    272.236111   976.514465   866.205807  480.832611
min     50.000000   142.000000   131.000000   70.000000
25%    146.250000   487.250000   437.250000  240.000000
50%    242.500000   832.500000   743.500000  410.000000
75%    338.750000  1177.750000  1049.750000  580.000000
max    435.000000  1523.000000  1356.000000  750.000000

In [31]:
Male = [50,142,131,70]
Female=[435,1523,1356,750]


In [32]:
Sales = [Male,Female]

In [33]:
Sales

[[50, 142, 131, 70], [435, 1523, 1356, 750]]

In [35]:
chiStats = sp.stats.chi2_contingency(Sales)
chiStats

(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

p = 0.66 >0.05 --> Accept Null Hypothesis
i.e,male-female buyer rations are similar across regions

# CustomerOrderForm.mtw


In [36]:
CustOF = pd.read_csv("C://Users//Lenovo//Downloads//Costomer+OrderForm.csv")

In [37]:
CustOF.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [39]:
CustOF.describe()

Phillippines   Indonesia       Malta       India
count           300         300         300         300
unique            2           2           2           2
top      Error Free  Error Free  Error Free  Error Free
freq            271         267         269         280

in phillipinees if 271 out of 300 order is error free, then 29 order is defective

in indonesia if 267 out of 300 order is error free, then 33 order is defective

in Malta if 269 out of 300 order is error free, then 31 order is defective

in India if 280 out of 300 order is error free, then 20 order is defective

In [40]:
chiStats = sp.stats.chi2_contingency([[271,267,269,280],[29,33,31,20]])
chiStats

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

p = 0.277>0.05, Hence we accept the null Hypothesis

proportion of defective % across the center is same.